apps/wallet/src/ui/app/components/accounts/VerifyPasswordModal.tsx

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { PasswordModalDialog, type PasswordModalDialogProps } from './PasswordInputDialog';
+
+type VerifyPasswordModalProps = Pick<PasswordModalDialogProps, 'open' | 'onClose'> & {
+	onVerify: (password: string) => Promise<void> | void;
+};
+
+export function VerifyPasswordModal({ onClose, onVerify, open }: VerifyPasswordModalProps) {
+	return (
+		<PasswordModalDialog
+			{...{
+				onClose,
+				open,
+				title: 'Verify your wallet password',
+				description: 'Enter your wallet password to authenticate',
+				verify: true,
+				confirmText: 'Verify',
+				cancelText: 'Cancel',
+				onSubmit: onVerify,
```

Comment:
saw your comment above, but I think it makes more sense to me to remove the `verify` prop from the `PasswordModalDialog` and move the verify logic to this component since it's specific to the `VerifyPasswordModal`

```tsx
onSubmit: async (password) => {
    try {
        await backgroundService.verifyPassword(password);
        onVerify(password)
    } catch (e) {
     // do something
    }
}
```

Reply:
Yeah that was the intention but then I had the issue that I couldn't set the wrong password error as a field error (don't have access to the form). So decided to keep it here in interest of time